Java/core/src/main/java/com/amazon/randomcutforest/RandomCutForest.java

```diff
@@ -1004,42 +1066,117 @@ public RandomCutForest build() {
-                .centerOfMassEnabled(forestState.centerOfMassEnabled).compactEnabled(forestState.compactEnabled)
-                .storeSequenceIndexesEnabled(forestState.storeSequenceIndexesEnabled)
-                .threadPoolSize(forestState.threadPoolSize)
-                .parallelExecutionEnabled(forestState.parallelExecutionEnabled).build();
-
-        if (!forestState.compactEnabled) {
+    public static RandomCutForest createForest(RandomCutForestState forestState) {
+        RandomCutForest forest = builder().dimensions(forestState.getDimensions())
+                .numberOfTrees(forestState.getNumberOfTrees()).lambda(forestState.getLambda())
+                .outputAfter(forestState.getOutputAfter()).sampleSize(forestState.getSampleSize())
+                .centerOfMassEnabled(forestState.isCenterOfMassEnabled()).compactEnabled(forestState.isCompactEnabled())
+                .storeSequenceIndexesEnabled(forestState.isStoreSequenceIndexesEnabled())
+                .threadPoolSize(forestState.getThreadPoolSize())
+                .parallelExecutionEnabled(forestState.isParallelExecutionEnabled()).build();
+
+        if (!forestState.isCompactEnabled()) {
             forest.updateExecutor.initializeModels(forestState.smallSamplerData, forestState.sequentialSamplerData);
         } else {
-            forest.pointStore.reInitialize(forestState.pointStoreDoubleData);
+            forest.pointStore.reInitialize(forestState.pointStoreDoubleState);
             if (forestState.saveTreeData) {
-                checkState(forestState.treeData != null, " error, missing tree information");
+                checkState(forestState.compactRandomCutTreeStates != null, " error, missing tree information");
             }
-            forest.updateExecutor.initializeCompact(forestState.compactSamplerData, forestState.treeData);
-
+            forest.updateExecutor.initializeCompact(forestState.compactSamplerStates,
+                    forestState.compactRandomCutTreeStates);
         }
 
-        forest.updateExecutor.setCurrentIndex(forestState.entreesSeen);
+        forest.updateExecutor.setCurrentIndex(forestState.getEntreesSeen());
         return forest;
     }
 
-    public ForestState getForestState() {
-        ForestState answer = new ForestState();
-        answer.numberOfTrees = getNumberOfTrees();
-        answer.dimensions = getDimensions();
-        answer.lambda = getLambda();
-        answer.sampleSize = getSampleSize();
-        answer.centerOfMassEnabled = centerOfMassEnabled();
-        answer.outputAfter = getOutputAfter();
-        answer.parallelExecutionEnabled = parallelExecutionEnabled();
-        answer.threadPoolSize = getThreadPoolSize();
-        answer.storeSequenceIndexesEnabled = storeSequenceIndexesEnabled();
-        answer.entreesSeen = getTotalUpdates();
-        answer.compactEnabled = compactEnabled();
+    public static class StateMapper implements IStateMapper<RandomCutForest, RandomCutForestState> {
```

Comment:
this mapper is nested.

Reply:
I'd like to leave this for now becase we haven't completed migrating the forest state class to the new paradigm and I'd like to keep those changes together. If I move this to the top level that will bring additional changes into this PR because the implementation code currently uses inner class visibility rules. Alternatively, I could delete this partial implementation from this PR.